### PR TITLE
feat: Add wellformedness checks and unchecked methods for floating-point operators

### PR DESCRIPTION
**Pull request by fabiomadge**
This PR adds safety checks to floating-point arithmetic and comparison operators while providing unchecked static methods as an escape hatch for edge case testing.

### Changes

**Wellformedness checks on operators:**
- Arithmetic operators (`+`, `-`, `*`, `/`) and comparisons (`<`, `<=`, `>`, `>=`) now require operands are not NaN
- Invalid infinity combinations are rejected: `∞+(-∞)`, `∞-∞`, `∞*0`, `0/0`, `∞/∞`
- Provides clear error messages indicating the specific violation

**Unchecked static methods:**
- Added 9 static methods per type: `Add`, `Sub`, `Mul`, `Div`, `Neg`, `Less`, `LessOrEqual`, `Greater`, `GreaterOrEqual`
- Follow IEEE 754 semantics exactly, including NaN propagation
- No preconditions - useful for testing edge cases

**Bug fix:**
- Fixed lazy loading of floating-point Boogie functions when types are inferred (e.g., `var x := fp64.PositiveInfinity`)
- Now works correctly with both legacy and PreType resolvers

### Example

```dafny
method SafeArithmetic(x: fp64, y: fp64) 
  requires !x.IsNaN && !y.IsNaN
  requires !(x.IsInfinite && y.IsInfinite && x.IsPositive != y.IsPositive)
{
  var sum := x + y;  // OK: preconditions prevent invalid operations
}

method EdgeCaseTesting() {
  var inf := fp64.PositiveInfinity;
  // var bad := inf - inf;           // ERROR: wellformedness check fails
  var result := fp64.Sub(inf, inf);  // OK: returns NaN per IEEE 754
  assert result.IsNaN;
}
```

### Testing
- All 21 fp32/fp64 tests passing
- Added comprehensive wellformedness test files
- Updated existing tests to use static methods for edge cases

### CODE REVIEW

File: docs/DafnyRef/Types.md [hunk 1]
```diff
@@ -436,7 +436,14 @@ Both `fp32` and `fp64` types support standard arithmetic operations following IE
 - Unary negation (`-`)
 - Comparisons (`<`, `<=`, `>`, `>=`)
 
-Note that due to floating-point rounding, familiar mathematical properties may not hold:
+**Well-formedness checks**: All these arithmetic operations and comparisons require that operands are not NaN. Additionally, certain combinations of infinity values produce invalid operations:
+
+- Addition: `∞ + (-∞)` is invalid
+- Subtraction: `∞ - ∞` is invalid
+- Multiplication: `∞ * 0` is invalid
+- Division: `0 / 0` and `∞ / ∞` are invalid
```
Comment (RustanLeino) [inline docs/DafnyRef/Types.md:444]: What about `x / 0`?

**Review by RustanLeino — APPROVED** [review]
Great job! Good test suite additions, too.

File: docs/DafnyRef/Types.md (hunk 1 above)
Comment (fabiomadge) [reply docs/DafnyRef/Types.md:444]: Also prohibited by the well-formedness checks
